enhancements/oc/mirroring-release-image-signatures.md

```diff
@@ -0,0 +1,256 @@
+1. Connect the removable media to a host within the air gapped environment that has access to both the central registry and to any cluster requiring upgrade.
+1. Upload the mirrored images to the central registry, as described in [the `--to-dir` flow](#pushing-to-disk):
+    ```console
+    $ oc image mirror "--from-dir=${REMOVABLE_MEDIA_PATH}/mirror" file://openshift/release:4.3.0* REGISTRY/REPOSITORY
+    ```
+1. Use `oc` to log in to a given cluster to be upgraded.
+1. Apply the the mirrored release image signature ConfigMap to the connected cluster, as described in [the `--to-dir` flow](#pushing-to-disk):
+    ```console
+    $ oc apply -f mirror/config/signature-sha256-81154f5c03294534.yaml
+    ```
+1. Perform the upgrade:
+    ```console
+    $ oc adm upgrade --allow-explicit-upgrade --to-image REGISTRY/REPOSITORY@sha256:81154f5c03294534e1eaf0319bef7a601134f891689ccede5d705ef659aa8c92
+    ```
+
+### Risks and Mitigations
+
+One risk considered is that a user familiar with the current behavior of the `oc adm release mirror ...` command may inadvertently apply configuration manifests to their cluster.
+This has been addressed by requiring the explicit `--apply-release-image-signature` argument to apply configuration manifests to a cluster.
+
+Another risk is that we will need to gather additional information from the wider internet, but Clayton feels like we are unlikely to ever have more than a handful of objects to capture all the information we need.
+This has been mitigated by using very specific options such as `--apply-release-image-signature` and `--release-image-signature-to-dir` to manage the release image signature ConfigMap.
+If `oc adm release mirror ...` learns about additional types in the future, it can add support for them with additional, specific, orthogonal options without confusion due to generic option names.
+
+## Design Details
+
+### Test Plan
+
+[The current e2e test][release-mirror-test] mirrors and then creates a target cluster from that mirror.
+It does not restrict access from the cluster to external networks, although work for that is [in flight][blackholed-proxy].
+The initial test implementation will cover:
+
+* Creating a cluster-under-test via the usual procedure, using [the *pushing to disk* flow](#pushing-to-disk) to provide an update, and then updating the cluster-under-test.
+    Once restricted-network proxy CI support is available, this test will be ported to use it, to ensure that all required content is being mirrored via the *pushing to disk* flow.
+* Similarly to the above, except using [the *applying directly to the target cluster* flow](#applying-directly-to-the-target-cluster), pushing both container images and configuration manifests into the cluster-under-test before updating it.
+
+### Graduation Criteria
+
+This will move straight to GA, because `oc` doesn't seem to have a history of a `--tech-preview` flag or similar to allow for graduated stability of new features.
```

Comment:
This seems weird and out of place.  I would rephrase this to be consistent with our general statement:

> This feature will be GA and will follow normal CLI policy for two releases of backwards compatibility.

Reply:
> This feature will be GA and will follow normal CLI policy for two releases of backwards compatibility.

I assume this is two minor bumps, like 4.5 -> 4.7.  But "falling out of GA support on the third minor bump" is distinct from "we are born in GA, because `oc` provides no mechanism to tech-preview new features".  I've pushed 528e93d -> 367e32d with an attempt to cover both sides.